hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/streaming/FileAndGrpcBlockItemWriter.java

```
@@ -52,14 +55,21 @@ public void writeItem(@NonNull byte[] bytes) {
     }
 
     @Override
-    public void writePbjItem(@NonNull BlockItem item) {
-        throw new UnsupportedOperationException("writePbjItem is not supported in this implementation");
+    public void closeCompleteBlock() {
+        this.fileBlockItemWriter.closeCompleteBlock();
```

Should each of these be wrapped in a try/catch so if one fails the other can be attempted? Speaking of... what happens if one succeeds but the other fails, is that OK?

---

I think you're right...we need to think very carefully in all of the `*BlockItemWriter` variants about where we would let an exception propagate.

That's a good topic to bring up with the team.